### PR TITLE
Support Python 3.6

### PR DESCRIPTION
**Pull request by hkvision**
## What changes were proposed in this pull request?

Test for python 3.6

## How was this patch tested?

Jenkins

## Related links or issues (optional)

#1293 #1665

### CODE REVIEW

File: pyspark/test/dev/prepare_env.sh [hunk 1]
```diff
@@ -33,7 +33,7 @@ export PYTHONPATH=$PYTHONPATH:$PYSPARK_ZIP:$DL_PYTHON_HOME:$DL_PYTHON_HOME/:$DL_
 export BIGDL_CLASSPATH=$(find $BIGDL_HOME/spark/dl/target/ -name "*with-dependencies.jar" | head -n 1)
 echo "BIGDL_CLASSPATH": $BIGDL_CLASSPATH
 
-export PYTHON_EXECUTABLES=("python2.7" "python3.5")
+export PYTHON_EXECUTABLES=("python2.7" "python3.6")
```
Comment (zhichao-li) [inline pyspark/test/dev/prepare_env.sh]: need to keep both python3.5 and python3.6

Comment (hkvision) [reply pyspark/test/dev/prepare_env.sh]: Yeah. This is only for test purpose. Will add it later.

**Comment by hkvision** [comment]
jenkins passed http://172.168.2.101:8080/view/PR-V/job/BigDL-PR-V/399/

**Comment by zhichao-li** [comment]
need to update the job for keras testing as well.

**Comment by hkvision** [comment]
@zhichao-li Yes. Will do it today. Currently in the nodes for keras, there is no spark version that works for python 3.6. We are now running on spark 2.1.0 and I will use spark 2.2.0 to test python 2.7, 3.5 and 3.6.

**Comment by hkvision** [comment]
@jason-dai have a look at the updated documentation for python 3.6?

File: docs/docs/PythonUserGuide/install-from-pip.md [hunk 2]
```diff
@@ -1,9 +1,10 @@
 ## **NOTES**
 
 - Pip install supports __Mac__ and __Linux__ platforms.
-- This package is built with JDK7 by default, so you need to install Java __>= JDK7__ before runing BigDL.
+- This package is built with __JDK7__ by default, so you need to install Java __>= JDK7__ before runing BigDL.
```
Comment (zhichao-li) [inline docs/docs/PythonUserGuide/install-from-pip.md]: install Java __>= JDK8__ before runing BigDL which is required by pyspark

**Comment by zhichao-li** [comment]
LGTM  Only with minor changes.

File: docs/docs/PythonUserGuide/install-from-pip.md (hunk 2 above)
Comment (hkvision) [reply docs/docs/PythonUserGuide/install-from-pip.md]: Modified.

**Comment by hkvision** [comment]
@jason-dai @zhichao-li Jenkins environment set up. For spark 2.1.1 and 2.2.0, testing for python 2.7, 3.5 and 3.6. For spark 1.5 and 1.6, testing for python 2.7 and 3.5.
For caffe-keras, using spark 2.1.1 and 2.2.0, testing for python 2.7, 3.5 and 3.6.

**Comment by hkvision** [comment]
jenkins passed http://172.168.2.101:8080/view/PR-V/job/BigDL-PR-V/406/

**Pull request merged**